lib/aika_web/plugs/api_auth_plug.ex

```diff
@@ -0,0 +1,22 @@
+defmodule AikaWeb.ApiAuthPlug do
+  import Plug.Conn
+  import Phoenix.Controller
+  alias Aika.{Repo, Accounts.User}
+
+  def init(default), do: default
+
+  def call(conn, _) do
+    %{"api_token" => api_token, "user_id" => user_id} = conn.params
+
+    user = Repo.get(User, user_id)
+
+    case user && user.api_token == api_token do
```

Comment:
Instead of the `case` you could use the `with` statement here to show the desired path. Something like this perhaps

```
with user <- Repo.get(User, user_id),
  ^api_token <- user.api_token do
      assign(conn, :user, user)
    else
    _ ->
      conn
      |> put_status(401)
      ...
    end
  end
end
```